packages/docs/docsite/src/components/splash/splash.tsx

```diff
@@ -0,0 +1,357 @@
+					)
+				}
+
+				// handle animation of background objects
+				for (let i = 0; i < bg.length; i++) {
+					const group = bg[i]
+					const children: HTMLCollection = group.children
+					for (let cIndex = 0; cIndex < children.length; cIndex++) {
+						const child = children[cIndex]
+						tl.from(
+							child,
+							0.8,
+							{
+								z: randomBetween(-1500, 2500),
+								x:
+									i % 2 === 0
+										? randomBetween(-80, -width)
+										: randomBetween(280, width / 2),
+								y: randomBetween(-height, height),
+								rotation: randomBetween(360, 720),
+								rotationX: randomBetween(-160, 160),
+								rotationY: randomBetween(-160, 160),
+								opacity: 0,
+								ease: Elastic,
+							},
+							'PieceTogether+=' + Math.random() * 0.3
+						)
+					}
+				}
+
+				tl.to(
+					'.background',
+					0.3,
+					{
+						opacity: 0,
+						easing: Elastic,
+						delay: 0.2,
+					},
+					'Scale' + 0
+				)
+
+				setTimeout(() => {
+					const svgElem = document.getElementById('svg-logo2')!
+					const cdims2 = svgElem.getBoundingClientRect()
+					const offset = document
+						.getElementById('chart-parts-letter-group')!
+						.getBoundingClientRect()
+					const callbackFunc = () => {
+						setAnimationOut(true)
+					}
+					const timeLineOut = new TimelineMax({
+						onComplete: callbackFunc,
+						delay: 0.1,
+					})
+					timeLineOut.addLabel('Scale', 0.4)
+					timeLineOut.addLabel('Links', 0.4)
+					timeLineOut.to(
+						svgElem,
+						0.4,
+						{
+							x: -(
+								cdims2.left +
+								(offset.left - cdims2.left) +
+								(cdims2.left + (offset.left - cdims2.left)) * 1.8
+							),
+							y: -(
+								cdims2.top +
+								(offset.top - cdims2.top) +
+								offset.height / 2 +
+								10
+							),
+							scale: 0.2,
+						},
+						'Scale'
+					)
+					timeLineOut.to(
+						title,
+						0.4,
+						{
+							opacity: 0,
+							delay: 0.35,
+						},
+						'Scale'
+					)
+					timeLineOut.to(
+						'.header',
+						0.1,
+						{
+							opacity: 1,
+							delay: 0.35,
+						},
+						'Scale'
+					)
+					// stagger in fade links right to left
+					const links = ['links', 'source', 'docs', 'blog']
+					links.forEach((item: string, index: number) => {
+						timeLineOut.to(
+							`#${item}`,
+							0.2,
+							{
+								opacity: 1,
+								delay: 0.4 + index * 0.1,
+							},
+							'Links+=' + 0
+						)
+					})
+
+					timeLineOut.play()
+				}, 1000)
+			}
+		}
+		animate()
+	}
+
+	// get all chart part elements
+	const getChartPartsChars = () => {
+		const chars = ['c', 'h', 'a', 'r', 't', 'p', 'a', 'r', 't', 's']
+		return chars.reduce((acc: HTMLElement[], letter: string, index: number) => {
+			const elem = document.getElementById(`char-${letter}${index + 1}`)!
+			acc.push(elem)
+			return acc
+		}, [])
+	}
+
+	// highlight div on mouseover
+	const mouseEnter = useCallback(
+		(event: React.MouseEvent<any>, ref: any) => {
```

Comment:
The render tree uses these functions inside of dynamic lambdas, which kind of negates their benefit. I think this would be better expressed using a custom hook function: e.g. 
```
const [blogMouseEnter, blogMouseLeave = usePaneMouseHandlers(blogRef)
const [docsMouseEnter, docsMouseLeave = usePaneMouseHandlers(docsRef)
```

Then you can pass the mouse handler functions into the render trees directly without defining dynamic functions every render.